### DISCUSSION TITLE
zh/patterns/abstract-factory/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#53] [Announcements]
# zh/patterns/abstract-factory/

或称 工具包 目的 提供一个用于创建相关对象家族的接口，而无需指定其具体类。 解释 真实世界例子 要创建一个王国，我们需要具有共同主题的对象。 精灵王国需要精灵王，精灵城堡和精灵军队，而兽人王国需要兽王，精灵城堡和兽人军队。 王国中的对象之间存在依赖性。 通俗的说 工厂的工厂； 一个将单个但相关/从属的工厂分组在一起而没有指定其具体类别的工厂。

https://java-design-patterns.com/zh/patterns/abstract-factory/

**Comment by SUN-SETS** [comment]
我的通俗理解是，假如一个公司中有不同的产品，不同的产品对应不同的工厂，其中不同产品针对于不同的合作方会有区别，比如说公司旗下的汽车产品，汽车产品对于不同合作方会有跑车，民用车等等的区别之分，所以说汽车产品也就对应的汽车工厂，以此类推，那么公司具有不同的产品也就会有不同的工厂，所以需要一个”工厂的工厂“ 来管理这些工厂，也就是说 在这个示例中 使用FactoryMaker 来根据不同产品的大类别 获取到对应的产品的工厂 ，然后用这个产品工厂来做实际的事情
